lib/internal/http2/compat.js

```diff
@@ -100,21 +100,20 @@ const assertValidHeader = hideStackFrames((name, value) => {
   }
 });
 
+const validPseudoHeaderNames = [
+  HTTP2_HEADER_STATUS,    // :status
+  HTTP2_HEADER_METHOD,    // :method
+  HTTP2_HEADER_PATH,      // :path
+  HTTP2_HEADER_AUTHORITY, // :authority
+  HTTP2_HEADER_SCHEME,    // :scheme
+];
+
 function isPseudoHeader(name) {
-  switch (name) {
-    case HTTP2_HEADER_STATUS:    // :status
-    case HTTP2_HEADER_METHOD:    // :method
-    case HTTP2_HEADER_PATH:      // :path
-    case HTTP2_HEADER_AUTHORITY: // :authority
-    case HTTP2_HEADER_SCHEME:    // :scheme
-      return true;
-    default:
-      return false;
-  }
+  return ArrayPrototypeIncludes(validPseudoHeaderNames, name);
```

Comment:
Please restore this. The switch is significant faster than using includes.

Reply:
or is it?

`default: 7.828ms`:
```js
function isPseudoHeader(name) {
  switch (name) {
    case HTTP2_HEADER_STATUS:
    case HTTP2_HEADER_METHOD:
    case HTTP2_HEADER_PATH:
    case HTTP2_HEADER_AUTHORITY:
    case HTTP2_HEADER_SCHEME:
      return true;
    default:
      return false;
  }
}

const name = ':scheme';

console.time();

for (let i = 0; i < 99999; i++) isPseudoHeader(name);

console.timeEnd();
```

`default: 4.235ms`:
```js
const validPseudoHeaderNames = [
  HTTP2_HEADER_STATUS,
  HTTP2_HEADER_METHOD,
  HTTP2_HEADER_PATH,
  HTTP2_HEADER_AUTHORITY,
  HTTP2_HEADER_SCHEME
];

function isPseudoHeader(name) {
  return validPseudoHeaderNames.includes(name);
}

const name = ':scheme';

console.time();

for (let i = 0; i < 99999; i++) isPseudoHeader(name);

console.timeEnd();
```

Tested in the latest Node.js release aka `v17.3.0`.

Reply:
I don't think a microbenchmark is particularly useful way to prove this given the various ways V8 can optimize things away, but also I'm getting faster results with switch when I use ':status' instead of ':scheme' and I get equivalent perf when using ':scheme'.

Even assuming perf is equivalent, what's the benefit of making the change?

Reply:
> Even assuming perf is equivalent, what's the benefit of making the change?

The change was mainly made for readability, it looks more readable than a `switch` statement with some cases in my opinion, but the `switch` statement might look more readable to others, wdyt?